pkg/hostpath/utils.go

```diff
@@ -125,3 +140,74 @@ func getVolumeDirectories(storagePoolDataDirs map[string]string) ([]string, erro
 	sort.Strings(directories)
 	return directories, nil
 }
+
+// RunPrometheusServer runs a prometheus server for metrics
+func RunPrometheusServer(metricsAddr string) {
+	reg := prometheus.NewRegistry()
+	reg.MustRegister(poolPathSharedWithOsGauge)
+	handler := promhttp.HandlerFor(reg, promhttp.HandlerOpts{})
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", handler)
+	server := http.Server{
+		Addr:    metricsAddr,
+		Handler: mux,
+	}
+
+	go func() {
+		err := server.ListenAndServe()
+		if err != nil {
+			klog.Error(err, "Failed to start Prometheus metrics endpoint server")
+		}
+	}()
+}
+
+func getMountInfos(args ...string) ([]MountPointInfo, error) {
+	cmdPath, err := exec.LookPath("findmnt")
+	if err != nil {
+		return nil, fmt.Errorf("findmnt not found: %w", err)
+	}
+
+	args = append(args, "--json")
+	out, err := exec.Command(cmdPath, args...).CombinedOutput()
+	if err != nil {
+		return nil, err
+	}
+
+	if len(out) < 1 {
+		return nil, fmt.Errorf("mount point info is nil")
+	}
+
+	mountInfos, err := parseMountInfo([]byte(out))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse the mount infos: %+v", err)
+	}
+
+	return mountInfos, nil
+}
+
+func checkVolumePathSharedWithOS(volumePath string) bool {
+	mountInfosForPath, err := getMountInfos("-T", volumePath)
+	if err != nil {
+		return false
+	}
+
+	mountInfosForCsiSockerDir, err := getMountInfos("-T", csiSockerDir)
+	if err != nil {
+		return false
+	}
+
+	if len(mountInfosForPath) != 1 || len(mountInfosForCsiSockerDir) != 1 {
+		return false
+	}
+
+	pathSource := mountInfosForPath[0].Source
+	csiSocketSource := mountInfosForCsiSockerDir[0].Source
+	if strings.Contains(pathSource, "[") {
```

Comment:
Maybe expand the mount info struct with something like this https://github.com/kubevirt/hostpath-provisioner-operator/blob/main/cmd/mounter/main.go#L335-L341 that includes some handling for special cases like RHCOS nodes.

Reply:
Actually I'm doing the same thing but uglier, thanks, I'll bring that in

Reply:
Done

Reply:
Actually, this grabs whatever is inside the square brackets (the path on host),
I only need whatever comes before the square brackets 
**/dev/vda1**[/var/hpvolumes] 
So the rhcos prefix is not a problem here

Reply:
Well the code I linked has to deal with both the path and the device part of the source, so code exists there to grab both. Unfortunately there is no standard library we can use so we have to roll our own. If you are only interested in the first part before the [ this will work.

Reply:
Can we have some unit tests to verify that it works?

Reply:
There are unit tests:
https://github.com/kubevirt/hostpath-provisioner/blob/1829b354cacba58a785fc842bebbcbef95927949/pkg/hostpath/hostpath_test.go#L119-L121